src/coreclr/inc/corinfoinstructionset.h

```diff
@@ -603,10 +603,14 @@ inline CORINFO_InstructionSetFlags EnsureInstructionSetFlagsAreValid(CORINFO_Ins
             resultflags.RemoveInstructionSet(InstructionSet_AVX512VBMI_VL);
         if (resultflags.HasInstructionSet(InstructionSet_AVX512VBMI_VL) && !resultflags.HasInstructionSet(InstructionSet_AVX512BW_VL))
             resultflags.RemoveInstructionSet(InstructionSet_AVX512VBMI_VL);
-        if (resultflags.HasInstructionSet(InstructionSet_VectorT128) && !resultflags.HasInstructionSet(InstructionSet_SSE2))
-            resultflags.RemoveInstructionSet(InstructionSet_VectorT128);
         if (resultflags.HasInstructionSet(InstructionSet_VectorT256) && !resultflags.HasInstructionSet(InstructionSet_AVX2))
+        {
+            // No AVX2 - downgrade to VectorT128
             resultflags.RemoveInstructionSet(InstructionSet_VectorT256);
+            resultflags.AddInstructionSet(InstructionSet_VectorT128);
+        }
+        if (resultflags.HasInstructionSet(InstructionSet_VectorT128) && !resultflags.HasInstructionSet(InstructionSet_SSE2))
+            resultflags.RemoveInstructionSet(InstructionSet_VectorT128);
```

Comment:
This file is autogenerated, so a manual fix wouldn't be correct.

I think the missing handling is in `codeman.cpp` when handling `CLRConfig::EXTERNAL_EnableAVX`

Reply:
> I think the missing handling is in codeman.cpp when handling CLRConfig::EXTERNAL_EnableAVX

I tried that first, just that what if `CLRConfig::EXTERNAL_EnableSSE41` is disabled so it imples AVX too - I can't check the knob directly I guess. Didn't realize this file is auto-generated

Reply:
@tannergooding pushed a better fix now